xbmc/utils/ActorProtocol.cpp

```diff
@@ -42,41 +43,40 @@ void Message::Release()
   payloadObj.release();
 
   // delete event in case of sync message
-  if (event)
-    delete event;
+  delete event;
 
-  origin->ReturnMessage(this);
+  origin.ReturnMessage(this);
 }
 
-bool Message::Reply(int sig, void *data /* = NULL*/, int size /* = 0 */)
+bool Message::Reply(int sig, void *data /* = NULL*/, size_t size /* = 0 */)
```

Comment:
I usually include `stddef.h` instead of `cstddef`; that guarantees the availability of `size_t` in the top-level namespace. But you're right, that's not 100% correct and not perfect C++.
Better would be to do a `using size_type = std::size_t;`(unless another class's `size_type` is available) inside the class, and use that type alias everywhere.
I'll write a patch to implement that, and I'll submit it in another PR. Just to have an example on how to do it properly in the Kodi code base.